estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,96 @@
+from datetime import timedelta
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+
+class estatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Propert offers"
+    _order = "price desc"
+
+    price = fields.Float(string="Price")
+    status = fields.Selection(
+        copy="False",
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused"),
+        ],
+    )
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(
+        string="Deadline",
+        compute="_compute_deadline",
+        inverse="_inverse_deadline",
+        store=True,
+    )
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    property_type_id = fields.Many2one(
+        related="property_id.property_type_id", store=True
+    )
+
+    _sql_constraints = [
+        ("check_offer_price", "CHECK(price > 0)", "Offer price must be > 0")
+    ]
+
+    @api.model_create_multi
+    def create(self, vals):
+        for record in vals:
+            property = record.get('property_id')
+            if property:
+                existing_offers = self.search([('property_id', '=', property)])
+                if any(offer.price >= record.get('price', 0) for offer in existing_offers):
+                    raise UserError("An existing offer has an equal or higher amount. Please submit a higher offer.")
+    
+            self.env['estate.property'].browse(record['property_id']).state = "recevied"
+        return super().create(vals)
+
+    @api.depends("validity", "create_date")
+    def _compute_deadline(self):
+        for offer in self:
+            if offer.create_date:
+                offer.date_deadline = offer.create_date.date() + timedelta(
+                    days=offer.validity
+                )
+            else:
+                offer.date_deadline = fields.Date.today() + timedelta(
+                    days=offer.validity
+                )
```

Comment:
avoid extra new lines
```suggestion
                offer.date_deadline = offer.create_date.date() + timedelta(days=offer.validity)
            else:
                offer.date_deadline = fields.Date.today() + timedelta(days=offer.validity)
```